### PR TITLE
Add Zhihu data (#1459)

### PR DESCRIPTION
**Pull request by wangrui6**
Adds Zhihu selected KOL data

Issue: #1459

### CODE REVIEW

**Comment by wangrui6** [comment]
Data scraping is still in progress and I plan to regularly update the Huggingface Dataset card here: https://huggingface.co/datasets/wangrui6/Zhihu-KOL

**Comment by olliestanley** [comment]
Is there a reason to include both the notebook and the main script? It seems to be the same code in two formats. I would suggest excluding the notebook from the repo for cleanliness

**Comment by wangrui6** [comment]
any update on this PR?

**Comment by wangrui6** [comment]
> Is there a reason to include both the notebook and the main script? It seems to be the same code in two formats. I would suggest excluding the notebook from the repo for cleanliness

deduped notebook and merged with mainline.

**Comment by Vechtomov** [comment]
Looks like something is wrong with your main merge. Can you recreate the pull request?

**Comment by wangrui6** [comment]
> Looks like something is wrong with your main merge. Can you recreate the pull request?

Rebased and squashed all the commits. Conflicts are also resolved.

**Comment by wangrui6** [comment]
@olliestanley @Vechtomov Can you review and leave any more comments if any?

File: data/datasets/zhihu-kol/main.py [hunk 1]
```diff
@@ -0,0 +1,244 @@
...
+    try:
+        uid = get_uid_by_url_token(url_token)
+    except Exception:
+        return pd.DataFrame(columns=operations.keys())
+    bar: tqdm = None
+    offset = 0
+    limit = 20
+    dfs: list[pd.DataFrame] = []
+    url = f"https://api.zhihu.com/members/{uid}/answers"
+    while 1:
```
Comment (bitplane) [inline data/datasets/zhihu-kol/main.py:97]: ought to be `while True`

**Comment by bitplane** [comment]
I haven't run it myself yet, but the code looks very nice. Well structured, isolated and documented, very readable. Nice work 🙂👍

**Review by Vechtomov — CHANGES_REQUESTED** [review]
Can you reupload the dataset? Looks like now `load_dataset("wangrui6/Zhihu-KOL")` loads only first datafile.
image

File: data/datasets/zhihu-kol/main.py (hunk 1 above)
Comment (wangrui6) [reply data/datasets/zhihu-kol/main.py:97]: will change it in the next PR.

**Comment by wangrui6** [comment]
> Can you reupload the dataset? Looks like now `load_dataset("wangrui6/Zhihu-KOL")` loads only first datafile. image

Fixed and merged into one file.

Question: In future, do we allow to load multiple parquet files? That will help people to manage the data version and not necessarily need to join data multiple times when there is new data obtained from crawlers. What do you think? 

Suggestion: allow `load_dataset("wangrui6/Zhihu-KOL")` to load multiple files in the data training pipeline.

**Review by Vechtomov — APPROVED** [review]

**Comment by Vechtomov** [comment]
> > Can you reupload the dataset? Looks like now `load_dataset("wangrui6/Zhihu-KOL")` loads only first datafile. image
> 
> Fixed and merged into one file.
> 
> Question: In future, do we allow to load multiple parquet files? That will help people to manage the data version and not necessarily need to join data multiple times when there is new data obtained from crawlers. What do you think?
> 
> Suggestion: allow `load_dataset("wangrui6/Zhihu-KOL")` to load multiple files in the data training pipeline.

`load_dataset` load all files by default. It seems that format of your first file broke this feature. Check this: https://huggingface.co/docs/datasets/repository_structure

**Review by bitplane — APPROVED** [review]
Looks good to me

**Comment by wangrui6** [comment]
Question, can we still constantly update the HuggingFace data card as we are crawling more and more data every day? Or should we create a new PR to give a different pointer in the data card list?

**Comment by Vechtomov** [comment]
> Question, can we still constantly update the HuggingFace data card as we are crawling more and more data every day? Or should we create a new PR to give a different pointer in the data card list?

Yes, you can update the dataset without new PR.

**Comment by wangrui6** [comment]
Update: Over 1 million highly voted question answer pairs have been uploaded to Huggingface https://huggingface.co/datasets/wangrui6/Zhihu-KOL/tree/main/data

**Pull request merged**